dsc_lib/src/configure/mod.rs

```diff
@@ -22,21 +24,21 @@ use security_context_lib::{SecurityContext, get_security_context};
 use serde_json::{Map, Value};
 use std::path::PathBuf;
 use std::{collections::HashMap, mem};
-use tracing::{debug, info, trace, warn_span, Span};
-use tracing_indicatif::span_ext::IndicatifSpanExt;
+use tracing::{debug, info, trace};
 pub mod context;
 pub mod config_doc;
 pub mod config_result;
 pub mod contraints;
 pub mod depends_on;
 pub mod parameters;
 
-pub struct Configurator {
+pub struct Configurator<'a> {
     json: String,
     config: Configuration,
     pub context: Context,
     discovery: Discovery,
     statement_parser: Statement,
+    progress_format: Option<&'a OutputFormat>,
```

Comment:
Instead of having this as an Option, I think it should just be `OutputFormat` and you have the constructor set a default.  This should also remove the need for lifetime.